### ISSUE TITLE
Requests proxied via YARP are missing in the traces

### ISSUE THREAD

**Issue by saithis** [getsentry/sentry-dotnet#3843]
### Package

Sentry

### .NET Flavor

.NET

### .NET Version

8.0.0

### OS

Any (not platform specific)

### SDK Version

4.13.0

### Steps to Reproduce

1. Create a ASP.NET Core project 
2. Add Sentry.AspNetCore and wire it up (.UseSentry + config)
```    
"Sentry": {
        "Dsn": "...",
        "IncludeActivityData": true,
        "MaxRequestBodySize": "None",
        "TracesSampleRate": 1.0
    },
``` 
3. Add Yarp.ReverseProxy
4. configure a route, that just echos the request back
5. configure a route, that takes the request and forwards it to the echo route via YARP (in production it would be another api for us, but for local debugging i tested it with the echo endpoint)
6. call the route via postman/etc.

 

### Expected Result

The proxied request should show up in the trace as a child 

### Actual Result

It doesn't
![Image](https://github.com/user-attachments/assets/64fc50f5-66c3-4bef-88a7-fba94684c995)

**Comment by bruno-garcia** [comment]
Is the proxy forwarding all headers? I imagine that might be the culprit.

Doesn't sound like a Sentry issue tbqh, there's something added in the middle that must be aware of headers and propagate them.

If the proxy is .NET, and runs Sentry, for example `Sentry.AspNetCore`, our SDK does the propagation automatically.

**Comment by Marioo1357** [comment]
@saithis When it started not work for you? From .Net 9 update or Sentry release?

**Comment by saithis** [comment]
@bruno-garcia YARP should be forwarding all headers, yes. Our app is .NET, uses Sentry via Sentry.AspNetCore and uses the YARP nuget package for proxying the requests.

@Marioo1357 We just started using Sentry and I'm just trying to get it to work. We had no previous version with sentry.

I tried it with Sentry.OpenTelemetry now. It shows the missing traces then, but some things appear twice:

![Image](https://github.com/user-attachments/assets/c5d4b3a5-40e5-4807-874a-9c4c5130288a)
Seems like every http.client request is recorded via Sentry.AspNetCore and via Sentry.OpenTelemetry

**Comment by bruno-garcia** [comment]
@jamescrosswell is the duplicated span when adding Sentry.OTel a known issue?

**Comment by jamescrosswell** [comment]
> [@jamescrosswell](https://github.com/jamescrosswell) is the duplicated span when adding Sentry.OTel a known issue?

It's not. @saithis how are you initializing Sentry and OpenTelemetry?

**Comment by saithis** [comment]
@jamescrosswell Here is the initialization code:
```
            builder.WebHost.UseSentry(c =>
            {
                c.Environment = builder.Environment.EnvironmentName;
                c.UseOpenTelemetry();
            });

            builder.Services.AddOpenTelemetry()
                .WithMetrics(b => b
                    .AddAspNetCoreInstrumentation()
                    .AddHttpClientInstrumentation()
                    .AddMeter("Microsoft.AspNetCore.Hosting", "Microsoft.AspNetCore.Server.Kestrel")
                    .AddView("request-duration",
                        new ExplicitBucketHistogramConfiguration
                        {
                            Boundaries = [0, 0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1, 2.5, 5, 7.5, 10],
                        })
                )
                .WithTracing(b => b
                    .SetResourceBuilder(ResourceBuilder.CreateDefault().AddService("API Gateway"))
                    .AddAspNetCoreInstrumentation(options =>
                        {
                            options.EnrichWithHttpRequest = EnrichActivityWithHttpRequest;
                        })
                    .AddHttpClientInstrumentation()
                    .AddSource("Yarp.ReverseProxy")
                    .AddJaegerExporter(options =>
                    {
                        options.AgentHost = settings.Instrumentation.JaegerHost;
                        options.AgentPort = settings.Instrumentation.JaegerPort;
                    })
                    .AddSentry()
                );
```

**Comment by jamescrosswell** [comment]
Thanks @saithis. 

The configuration looks all good.

If you comment out the line `AddSource("Yarp.ReverseProxy")` do you still get duplicate traces for requests?

**Comment by saithis** [comment]
@jamescrosswell Yeah, they are still doubled. Only difference between the results is, that the "(3^) proxy.forwarder - proxy forwarder" line is not there anymore. 

Also: The request with the guid in it is the only one done by the YARP proxy part and is not doubled. 
All other requests are done via either Microsoft.AspNetCore.Authentication.JwtBearer or IdentityModel.Client nuget packages, which I assume to use HttpClient and they are all doubled.

**Comment by saithis** [comment]
https://github.com/saithis/sentry-issue-reproduction

![Image](https://github.com/user-attachments/assets/99fae0ac-4faa-4422-a676-5e18032f3b76)

For me it seems like sentry is recording the request high up in the HttpClients message handler chain and OTEL is doing it again on a lower level. That could be the reason why sentry only records google.com and OTEL also records the redirect from google.com to consent.google.com.

But strangely it is not double recorded after the proxy called itself again on the special debug route.

**Comment by jamescrosswell** [comment]
Thanks for the reproduction! ❤ 

This isn't related to YARP... It's a result of having multiple pieces of code instrument outbound HTTP requests. 

[Your code](https://github.com/saithis/sentry-issue-reproduction/blob/b215c684135ebfdd5c60a029d4cc93bafea421ef/SentryIssue3843/Program.cs#L31) calls:
```csharp
    .AddHttpClientInstrumentation()
```

That tells OpenTelemetry to create trace spans for outbound HTTP Requests. 

Additionally, Sentry's ASP.NET Core integration registers a `SentryHttpMessageHandler`:
https://github.com/getsentry/sentry-dotnet/blob/dcc32b22b29fe77f1129f9cb848388af6093ece5/src/Sentry.Extensions.Logging/SentryHttpMessageHandlerBuilderFilter.cs#L17-L22

Sentry's handler both creates traces for outbound requests and (optionally) creates issues for failed requests. 

I think we should add an option to allow SDK users to disable Sentry's `HttpMessageHandler`, for users that would rather use the OpenTelemetry package for this.

@saithis as a workaround, in the short term, you could comment out the call to `.AddHttpClientInstrumentation()` so that there is only one library trying to create traces for outbound HTTP reqeuests.

**Comment by saithis** [comment]
Thank you, I'll try that :)

**Comment by jamescrosswell** [comment]
@saithis FYI an option to disable Sentry's buildin `SentryHttpMessageHandler` has been added in:
- https://github.com/getsentry/sentry-dotnet/pull/3879

This will be included in the next release, which gives you another option (you can either disable Sentry's HttpClient instrumentation or disable the OpenTelemetry one, to prevent duplicate instrumentation of outbound HTTP requests).
